neps/nep-0492.md

```diff
@@ -0,0 +1,74 @@
+---
+NEP: 492
+Title: Restrict creation of Ethereum Addresses
+Authors: Bowen Wang<bowen@near.org>
+Status: Draft
+DiscussionsTo: https://github.com/near/NEPs/pull/492
+Type: Protocol
+Version: 0.0.0
+Created: 2023-07-27
+LastUpdated: 2023-07-27
+---
+
+## Summary
+
+This proposal aims to restrict the creation of top level accounts (other than implicit accounts) on NEAR to both prevent loss of funds due to careless user behaviors and scams
+and create possibilities for future interopability solutions.
+
+## Motivation
+
+Today an [Ethereum address](https://ethereum.org/en/developers/docs/accounts/) such as "0x32400084c286cf3e17e7b677ea9583e60a000324" is a valid account on NEAR and because it is longer than 32 characters,
+anyone can create such an account. This has unfortunately caused a few incidents where users lose their funds due to either a scam or careless behaviors.
+For example, when a user withdraw USDT from an exchange to their NEAR account, it is possible that they think they withdraw to Ethereum and therefore enter their Eth address.
+If this address exists on NEAR, then the user would lose their fund. A malicious actor could exploit this can create known Eth smart contract addresses on NEAR to trick users to send tokens to those addresses. With the proliferation of BOS gateways, including Ethereum ones, such exploits may become more common as users switch between NEAR wallets and Ethereum wallets (mainly metamask).
+
+In addition to prevent loss of funds for users, this change allows the possibility of Ethereum wallets supporting NEAR transactions, which could enable much more adoption of NEAR. The exact details of how that would be done is outside the scope of this proposal.
+
+There are currently ~5000 Ethereum addresses already created on NEAR. It is also outside the scope of this proposal to discuss what to do with them. 
```

Comment:
What about other non-Ethereum-like top level accounts whos creation would have been restricted by this proposal?

Reply:
There are a total of 11423 accounts on mainnet that are top-level accounts longer than 32 characters and that are not implicit accounts. I don't think there is a clear pattern that people use top-level accounts longer than 32 characters for.